search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java

```diff
@@ -161,22 +165,39 @@ private BoolQueryBuilder boolQuery(ResourceAwsQuery query) {
                 var searchFields = key.searchFields().toArray(String[]::new);
                 if (hasMultipleFields(searchFields)) {
                     bq.must(QueryBuilders
-                                .multiMatchQuery(escapeSearchString(value), searchFields)
-                                .operator(Operator.AND));
+                        .multiMatchQuery(escapeSearchString(value), searchFields)
+                        .operator(Operator.AND));
+                    if (key.equals(CONTRIBUTOR)) {
+                        addPromotedQuery(query, bq);
+                    }
                 } else {
                     var searchField = searchFields[0];
                     if (isRangeQuery(key)) {
                         bq.must(rangeQuery(key.searchOperator(), searchField, value));
                     } else {
                         bq.must(QueryBuilders
-                                    .matchQuery(searchField, escapeSearchString(value))
-                                    .operator(Operator.AND));
+                            .matchQuery(searchField, escapeSearchString(value))
+                            .operator(Operator.AND));
+                        if (key.equals(CONTRIBUTOR)) {
+                            addPromotedQuery(query, bq);
+                        }
                     }
                 }
             });
         return bq;
     }
 
+    private void addPromotedQuery(ResourceAwsQuery query, BoolQueryBuilder bq) {
+        var promotedPublications = userSettingsClient
+            .doSearch(query)
+            .promotedPublications();
+        for (int i = 0; i < promotedPublications.size(); i++) {
+            bq.should(QueryBuilders.matchQuery("id", promotedPublications.get(i))
+                .boost(promotedPublications.size() - i));
+        }
+        query.removeKey(SORT);
```

Comment:
why is this here?

Reply:
In the code I borrow from, it doesn't run if sort is set. SORT is always sat in the querybuilder, so I have to remove it in order to make this work, is that is how I interpret the code.

Reply:
maybe that should be described in the code so other people know why? either as describing code or a comment..?